src/staticdata.c

```diff
@@ -1438,10 +1442,14 @@ static void jl_write_values(jl_serializer_state *s) JL_GC_DISABLED
                 if (dt->layout != NULL) {
                     size_t nf = dt->layout->nfields;
                     size_t np = dt->layout->npointers;
-                    size_t fieldsize = jl_fielddesc_size(dt->layout->fielddesc_type);
+                    size_t fieldsize = 0;
+                    uint8_t is_foreign_type = dt->layout->fielddesc_type == 3;
+                    if (!is_foreign_type) {
```

Comment:
I don't really understand the relation between the code in dump.c and staticdata.c, so this may be incredibly stupid, but: is it perhaps necessary to make similar changes in `dump.c` ?

Reply:
The parent branch removes `dump.c` completely.

Reply:
Ah great!